FRAuth/FRAuth/Callbacks/CallbackFactory.swift

```diff
@@ -81,9 +81,7 @@ public class CallbackFactory: NSObject {
             CallbackType.WebAuthnRegistrationCallback.rawValue: WebAuthnRegistrationCallback.self,
             CallbackType.IdPCallback.rawValue: IdPCallback.self,
             CallbackType.SelectIdPCallback.rawValue: SelectIdPCallback.self,
-            CallbackType.DeviceBindingCallback.rawValue: DeviceBindingCallback.self,
-            CallbackType.DeviceSigningVerifierCallback.rawValue: DeviceSigningVerifierCallback.self
-        ]
+        ] 
```

Comment:
For add the callback dynamically, its going to be tricky because we cannot import FRDeviceBind module in FRAuth that would lead to a Circular dependency between modules. so far i can think of couple of ideas

1. So my first idea is in FRUSerkey class , which is public facing , but this can be changed and let the user to call the register from their original application, which is kind of work around 

class FRUSerkey {

 public func register() {
        CallbackFactory.shared.registerCallback(callbackType: "DeviceBindingCallback", callbackClass: DeviceBindingCallback.self)
        CallbackFactory.shared.registerCallback(callbackType: "DeviceSigningVerifierCallback", callbackClass: DeviceSigningVerifierCallback.self)
    }
}


2. This approach will work for sure , but will require lot of work.
How about DeviceBind module to just depend only on FRCore Module? so that we can fix the circular dependency issue. and we can easily do this

#if canImport(FRDeviceBinding)
    import FRDeviceBinding
#endif
 


This is the easiest approach , How about we add a method in FRDeviceBind Module to register

Reply:
+1

Reply:
@jeyanthanperiyasamy I thought about the first approach. Probably the best is to create a new class with the same name as the module (FRDeviceBinding) and add the register method there. 

As for the second one, Yes you are right, I tried it with FRAuth and ran into circular dependency issue. I am not sure we will be able to import just FRCore, as FRDeviceBinding has so many dependencies from FRAuth. Let's connect and discuss this one.

Reply:
Is there a way to check if the class or framework exists?
https://stackoverflow.com/questions/42099811/how-to-check-if-ios-application-uses-storekit-framework